### PR TITLE
Create native interop section

### PR DESCRIPTION
**Pull request by BillWagner**
Contributes to #33878 

Move the articles on interop into the advanced topics section.

### CODE REVIEW

**Comment by BillWagner** [comment]
Once again, this is a larger PR in terms of the number of files. However, not much changed in the files or the code samples. I did run Acrolinx and fix many of those style issues.

Beyond that, this moved files, updated links, and created the new section.

One area that should be reviewed: 

Two walkthroughs dated from the time when C# and VB were "co-evolving". There was a single article that mixed both C# and VB samples for dynamic programming. In both cases, I split that article in two, creating a VB and a C# version of each article. Links are here:

- [Internal preview - Creating and using dynamic objects in C#](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/interop/walkthrough-creating-and-using-dynamic-objects?branch=pr-en-us-34132)
- [Internal preview - Creating and using dynamic objects in VB](https://review.learn.microsoft.com/en-us/dotnet/visual-basic/programming-guide/language-features/early-late-binding/walkthrough-creating-and-using-dynamic-objects?branch=pr-en-us-34132)
- [Internal preview - Office programming in C#](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/interop/walkthrough-office-programming?branch=pr-en-us-34132)
- [Internal preview - Office programming in VB](https://review.learn.microsoft.com/en-us/dotnet/visual-basic/programming-guide/com-interop/walkthrough-office-programming?branch=pr-en-us-34132)

I didn't run Acrolinx on the VB samples.

**Comment by BillWagner** [comment]
@adegeo 

The one snippets 5000 error in this PR is odd to me. It's the Office Interop VSTO project. Thecontainer doesn't have that workload, so it's an expected failure. I've added that to the expected errors page, but it's not working correctly.

File: docs/csharp/advanced-topics/interop/walkthrough-office-programming.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+This walkthrough demonstrates these features in the context of Office programming, but many of these features are also useful in general programming. In the walkthrough, you use an Excel Add-in application to create an Excel workbook. Next, you create a Word document that contains a link to the workbook. Finally, you see how to enable and disable the PIA dependency.
+
+## Prerequisites
+
+You must have Microsoft Office Excel and Microsoft Office Word installed on your computer to complete this walkthrough.
+
+[!INCLUDE[note_settings_general](~/includes/note-settings-general-md.md)]
+
+### To set up an Excel Add-in application
```
Comment (adegeo) [inline docs/csharp/advanced-topics/interop/walkthrough-office-programming.md:25]: I think you could easily trim these headers (and the ones in the VB article) to ## and get them in the TOC. Right now the TOC is simply:

- Prerequisites
- See Also

File: docs/csharp/advanced-topics/interop/snippets/OfficeWalkthrough/snippets5000.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (adegeo) [inline docs/csharp/advanced-topics/interop/snippets/OfficeWalkthrough/snippets5000.json:1]: This file name is incorrect. It should be:

snippets`.`5000.json

**Review by adegeo — APPROVED** [review]
Looks good to me. Approving to unblock.

- One minor suggestion with headers
- One fix for snippets build.

**Comment by BillWagner** [comment]
closing and reopening for a fresh CI build

**Pull request merged**